docs/proposal-hosting-option.md

```diff
@@ -0,0 +1,75 @@
+
+By separating the publication ideas from the content ideas (and making them interoperable) I think we'd end up with much more cool ideas/experimentation than tying them together.
+
+## Ideal architecture
+
+* A user on GitHub (`<username>`) creates a public Foam repo name `foam-notes` (or something standardised)
+* The URL `<username>.foam.website` (or something similar) is immediately available as a public view of their notes.
+* Behind the scenes, by visiting a new username, the metadata for building up backlinks etc is built up in the background.
+* Once the site has been crawled & cached, performance should be indistinguishable from self-hosted static files.
+* Content changes are propagated within seconds with clever background fetching & caching
+
+For non-GitHub users, it would be easy to set up the same sort of behaviour under a URL scheme `<gitlab-username>.gitlab.foam.website`.
+
+## How
+
+I could give a technical pitch for why I want to build it this way, but the truth is: I want to build it as a tech demo/showcase of what [FABs](https://fab.dev) can do running on an edge, using [Cloudflare Workers](https://workers.dev), deployed using [Linc](https://linc.sh). So, my proposal:
+
+* NextJS in server-rendered mode
+* Hosted in a FAB in Cloudflare Workers
+* Using the [KV store](https://developers.cloudflare.com/workers/reference/storage) as a global cache for metadata/rendered documents.
+
+For the first request for any new URL:
+
+* Fetch the raw markdown from GitHub directly
+* Store it in the Cache
+* Return rendered page
+
+For repo-wide metadata like backlinks, we don't need to slow down the first request, but can set off a crawler process in the background.
+
+Both of these processes lend themselves to a "serve stale" approach:
+
+* The edge renderer _always_ responds with the current version in-cache, so responses are always fast (after the very first requests).
+* Periodically (ideally, once every ~5-10 seconds) GitHub is checked to see if there have been updates. If there have been, the cache is refreshed and new requests will get the updated content. **This means the time between pushing new content and it being visible could be as low as 5 seconds**.
+
+## Alternatives
+
+This is the way I'd like to build it, but it's not the only approach:
+
+* A static-site generator could be part of the Foam workspace example (as it sorta is now). People can publish using GitHub pages or Netlify (or Linc & Cloudflare, too).
+	* ❌ Needs users to register a separate account with another service (assuming GitHub pages isn't sufficient)
+	* ✅ A user's repo is wholly self-contained
+	* ❌ Having the SSG bundled means it will gradually diverge between users, which might make new features harder for people to adopt, and make it less possible for people to share ideas.
```

Comment:
That works for a central maintainer pushing updates, but doesn't let people try out someone else's renderer with their content without pulling/forking/copying etc, so I think it'd be less of an enticement for exploration.

A central authority releasing updates to the `.vscode` directory does make a fair bit of sense though, since that's kind of what Foam _is_.

Reply:
> doesn't let people try out someone else's renderer with their content without pulling/forking/copying etc, so I think it'd be less of an enticement for exploration

Possibly, but getting your renderer on the service would also be a lot more difficult than just pushing a template on GH and so the barrier would be a lot higher?

Reply:
Mayyyybe? But the idea that `<any-github-username>.my-fork-of-foam-renderer.com` could be made to work means that you could try anything out you like?

Forking the renderer code and deploying it would be a little bit more work than using GH pages, sure, but basically the same number as steps as setting up any Netlify site.

Reply:
> Forking the renderer code and deploying it would be a little bit more work than using GH pages, sure, but basically the same number as steps as setting up any Netlify site

Is it really? I mean I guess it depends how the renderer works but imagine I'm creating a shiny new framework in rust and implement a foam renderer with it. How do I plug it in?